iaso/click/lazy.py

```diff
@@ -0,0 +1,71 @@
+import importlib
+import re
+
+import click
+
+IMPORT_PATTERN = re.compile(
+    r"from\s+(\.*(?:[a-zA-Z_][a-zA-Z0-9_]*)?(?:\.(?:[a-zA-Z_][a-zA-Z0-9_]*))*)\s+import\s+(?:\(\s*)?((?:(?:[a-zA-Z_][a-zA-Z0-9_]*)(?:\s+as\s+[a-zA-Z_][a-zA-Z0-9_]*)?(?:,\s*)?)+)\s*\)?"
+)
+
+
+def LazyCommandGroup(package, subcommands):
```

Comment:
Let's have a chat on this in the next office day, although it may not look a very Pythonist way of doing things... maybe it's justified.

Reply:
For click, this is the suggested way to do dynamic loading of commands (see https://click.palletsprojects.com/en/7.x/commands/#custom-multi-commands) - I just modified it so that it knows its subcommands in advance. The reason for lazily loading in subcommands is to improve the startup time of the overall command. As Python just loads every dependency otherwise, we need to add barriers so it only loads in modules as absolutely necessary.